pkg/nuctl/command/deploy.go

```diff
@@ -778,13 +781,44 @@ func (d *deployCommandeer) patchFunction(ctx context.Context, function string) e
 	requestHeaders := d.resolveRequestHeaders()
 
 	if err := d.betaCommandeer.apiClient.PatchFunction(ctx,
-		function,
+		functionName,
 		d.rootCommandeer.namespace,
 		payload,
 		requestHeaders); err != nil {
 		return errors.Wrap(err, "Failed to patch function")
 	}
 
+	d.rootCommandeer.loggerInstance.InfoWithCtx(ctx,
+		"Function redeploy request sent successfully",
+		"function", functionName)
+
+	if d.waitForFunction {
+		ticker := time.NewTicker(5 * time.Second)
+		for {
+			select {
+			case <-time.After(d.waitTimeout):
+				return errors.New(fmt.Sprintf("Timed out waiting for function '%s' to be ready", functionName))
+			case <-ticker.C:
+
+				// get function and poll its status
+				function, err := d.betaCommandeer.apiClient.GetFunction(ctx, functionName, d.rootCommandeer.namespace)
+				if err != nil {
+					d.rootCommandeer.loggerInstance.WarnWithCtx(ctx, "Failed to get function", "functionName", functionName)
+				}
+				if function.Status.State == functionconfig.FunctionStateReady {
+					d.rootCommandeer.loggerInstance.InfoWithCtx(ctx,
+						"Function redeployed successfully",
+						"functionName", functionName)
+					return nil
+				}
+				d.rootCommandeer.loggerInstance.DebugWithCtx(ctx,
+					"Function not ready yet",
+					"functionName", functionName,
+					"functionState", function.Status.State)
+			}
+		}
+	}
```

Comment:
Separate this out to 2 extra methods:
- functionIsReady
- waitForFunctionDeployment (which uses functionIsReady)
And then use waitForFunctionDeployment here

Reply:
also check all terminal states